### ISSUE TITLE
Working?

### ISSUE THREAD

**Issue by SerialFX** [Romain-P/alpharite#2]
Loaded into VS19 and built out AlphaRite only writes class DLLs to bin and no executable - could you include that part please?

**Comment by Romain-P** [comment]
alpharite compiles on my side. but offsets changed since they recently did an update to MergedUnity.dll 
I'm effraid the sdk wouldnt work anymore

**Comment by Naturious** [comment]
How can I work on this? Can you help me just to onboard and I'll take it from there?

**Comment by Romain-P** [comment]
You gotta de-obfuscate `MergedUnity.dll` using https://github.com/de4dot/de4dot
Once you have a clean `MergedUnity.dll`, you gotta use https://github.com/CabbageCrow/AssemblyPublicizer in order to make public all internal classes/namespaces so then you can access them

Then import use this mergedunity as reference for this project, and compile it.
Then either inject the dll with any injector or load the dll from any method inside MergedUnity.dll and replace it in the data folder. (both alpharite.dll & mergedunity.dll)

**Comment by Naturious** [comment]
Just compiling de4dot took me 30 minutes and I can't even get it right, and there are no binaries hosted anymore, any ideas?
I assume I de-obfuscate the `MergedUnity.dll` that's already in my battlerite folder?

**Comment by Romain-P** [comment]
yes basically you just have to drag & drop `MergedUnity.dll` on de4dot.exe and the cleaned dll should be automatically added on the same folder

**Comment by Naturious** [comment]
I did all that and when I copy MergedUnity and the compiled Alpharite.dll to the data folder it crashes the game on start

**Comment by Naturious** [comment]
In fact the publicized assembly is the one that's crashing the game, even without compiling it on my end

**Comment by Romain-P** [comment]
I did it from my side, check my last commit, `MergedUnity.dll` is updated

**Comment by Naturious** [comment]
The new DLLs crash battlerite, just like the one I created, am I missing something in the injection process?
Shouldn't the game work if I replace the DLLs in /lib to Battlerite/Battlerite_Data/Managed?

**Comment by obabac** [comment]
It works for me. Just copied MergedUnity.dll and AlphaRite.dll into Battlerite/Battlerite_Data/Managed.

**Comment by Limalimao** [comment]
the aimbot doesnt work, the menu opens and the wallhack works but the aimbot doesnt work for me, i have to press a key to work?

**Comment by Limalimao** [comment]
> It works for me. Just copied MergedUnity.dll and AlphaRite.dll into Battlerite/Battlerite_Data/Managed.

the aimbot works for you?

**Comment by Naturious** [comment]
Alright works for me, I had to reinstall, thanks!

**Comment by obabac** [comment]
> > It works for me. Just copied MergedUnity.dll and AlphaRite.dll into Battlerite/Battlerite_Data/Managed.
> 
> the aimbot works for you?

If the aimbot dosn't work, recompile the project and copy AlphaRite.dll from the builds to Battlerite/Battlerite_Data/Managed

**Comment by Limalimao** [comment]
![Screenshot_44](https://user-images.githubusercontent.com/45106054/111658017-babd5780-87ea-11eb-96df-6afebf13ab8b.png)
![Screenshot_45](https://user-images.githubusercontent.com/45106054/111658020-bb55ee00-87ea-11eb-913b-a871c2277b5c.png)
![Screenshot_46](https://user-images.githubusercontent.com/45106054/111658021-bbee8480-87ea-11eb-8e8f-b0dd04ec3b47.png)
![Screenshot_47](https://user-images.githubusercontent.com/45106054/111658022-bbee8480-87ea-11eb-9a35-b1da1265bc50.png)


> > > It works for me. Just copied MergedUnity.dll and AlphaRite.dll into Battlerite/Battlerite_Data/Managed.
> > 
> > 
> > the aimbot works for you?
> 
> If the aimbot dosn't work, recompile the project and copy AlphaRite.dll from the builds to Battlerite/Battlerite_Data/Managed

is this how it works? i dont code, i dont know this things, maybe the aimbot is working and i dont know the aimkey? this is how the menu looks for me 
![Screenshot_43](https://user-images.githubusercontent.com/45106054/111658363-0b34b500-87eb-11eb-849c-782b164b38f4.png)

**Comment by Romain-P** [comment]
Indeed `Alpharite.dll` was an old release. I recompiled the dll, the last release is again inside `/releases`. Take this one and relaunch the game, you should see the neew menu
edit: press INSERT ingame for opening the menu

**Comment by Limalimao** [comment]
> Indeed `Alpharite.dll` was an old release. I recompiled the dll, the last release is again inside `/releases`. Take this one and relaunch the game, you should see the neew menu
> edit: press INSERT ingame for opening the menu

nice man, now works fine S2

**Comment by darkno89** [comment]
hi can you please re-recompilate the alpharite.dll ?. Seems to be laggy when we play ! ty so much !

**Comment by Romain-P** [comment]
> hi can you please re-recompilate the alpharite.dll ?. Seems to be laggy when we play ! ty so much !

Since battlerite made another update I guess that's why

**Comment by Naturious** [comment]
I think it's just because the default button for aim lock is W so when you don't change it when you go into a game it laggs so much for not finding a target.
Everything works fine on my end still.

**Comment by Romain-P** [comment]
@Naturious @darkno89 @obabac @Limalimao 
I've worked a bit on it since recent events on BR.
I'm offering a private version with an advanced aim prediction (more accurate depending the champion)
Also some scripts have been added (auto counters against melees (before the cast being finished) and also against casters). There might be more in the future

Let me know if someone would be interesting to buy this.
Discord/ `tricky#0833`
